### PR TITLE
fix: Stop duplication locations within the DB #4930

### PR DESCRIPTION
**Pull request by laf**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Removes duplicates and then creates a unique index on lat,lng.

Moved location lookup to discovery to save lookups on busy networks, I don't think it's a concern that we don't pick the change up within 5 minutes.

### CODE REVIEW

**Comment by towster** [comment]
I have locations that are physically same lat/lng but with different names. For example a building with different floors. I think from reading this that it would not allow me to have two locations with same lat,lng.

**Comment by laf** [comment]
We only do the following lookup: SELECT `lat`,`lng` FROM `locations` WHERE `location`=? LIMIT 1 which means that if lat/lng are seen already then we don't do another lookup which means this change doesn't affect how things work already

**Comment by towster** [comment]
So the schema change will delete 144 of my locations. 

```
MariaDB [librenms]> select count(distinct(l1.id)) FROM `locations` `l1`, `locations` `l2` WHERE `l1`.`lat` = `l2`.`lat` AND `l1`.`lng` = `l2`.`lng` AND `l1`.`id` > `l2`.`id`;
+------------------------+
| count(distinct(l1.id)) |
+------------------------+
|                    144 |
+------------------------+

```

File: sql-schema/174.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+DELETE `l1` FROM `locations` `l1`, `locations` `l2` WHERE `l1`.`lat` = `l2`.`lat` AND `l1`.`lng` = `l2`.`lng` AND `l1`.`id` > `l2`.`id`;
+ALTER TABLE `locations` ADD UNIQUE( `lat`, `lng`);
```
Comment (geordish) [inline sql-schema/174.sql]: Based on @towster's comments, should this be perhaps:

```SQL
DELETE `l1` FROM `locations` `l1`, `locations` `l2` WHERE `l1`.`lat` = `l2`.`lat` AND `l1`.`lng` = `l2`.`lng` AND `l1`.`location` = `l2`.`location` AND `l1`.`id` > `l2`.`id`;
ALTER TABLE `locations` ADD UNIQUE( `lat`, `lng`, `location`);
```

Comment (towster) [inline sql-schema/174.sql]: Much better - I can have a building with multiple floors again..

```
MariaDB [librenms]> select count(distinct(l1.id)) FROM `locations` `l1`, `locations` `l2` WHERE `l1`.`lat` = `l2`.`lat` AND `l1`.`lng` = `l2`.`lng` AND `l1`.`location` = `l2`.`location` AND `l1`.`id` > `l2`.`id`;
+------------------------+
| count(distinct(l1.id)) |
+------------------------+
|                      0 |
+------------------------+
```

**Pull request closed without merging**